src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs

```diff
@@ -277,8 +277,21 @@ public static bool IsInLeftOfDeconstructionAssignment(this IOperation operation,
             return false;
         }
 
+        public static bool IsOutArgument(this IOperation operation, [NotNullWhen(true)] out IInvocationOperation? invocation)
+        {
+            invocation = null;
+
+            if (operation.Parent is IArgumentOperation argumentOperation &&
+                    argumentOperation.Parameter?.RefKind == RefKind.Out)
+            {
+                invocation = argumentOperation.Parent as IInvocationOperation;
+            }
+
+            return invocation is not null;
```

Comment:
```suggestion
            if (operation.Parent is IArgumentOperation argumentOperation &&
                    argumentOperation.Parameter?.RefKind == RefKind.Out)
            {
               return argumentOperation.Parent is IInvocationOperation;
            }

            return false;
```

Reply:
@DoctorKrolic `invocation` needs to be assigned as it's an `out` parameter. The suggestion doesn't assign it.